### DISCUSSION TITLE
Best way to redeploy using non-root user for first time?

### DISCUSSION THREAD

**Discussion by luke-gru** [basecamp/kamal#1188] [General]
Hi, first off thanks for making kamal, it's very useful. I was playing around with kamal and deployed my app as root user but once I got it working I now want to lock down ssh on the server by not allowing root logins anymore. What's the best way to redeploy using a non-root user? (apart from of course specifying the user in the `ssh` section of the yml file). Do I need to remove the containers, and then `kamal setup` again once I change config/deploy.yml? I'm looking for a list of commands I should run. Thanks!

**Comment by tuladhar** [comment]
Apart from specifying the SSH login user in deploy config, you want to make sure non-root user is assigned to docker group as [mentioned here](https://kamal-deploy.org/docs/configuration/ssh/).

Once, above is configured, you can skip `kamal setup` and run `kamal deploy` as usual.
